src/views/GraphDetail.vue

```diff
@@ -427,19 +427,22 @@ export default Vue.extend({
         offset: this.offset,
         limit: this.limit,
       });
+      // eslint-disable-next-line no-underscore-dangle
+      const edges = await api.edges(this.workspace, this.graph, nodes.results[0]._id, {
+        offset: this.offset,
+        limit: this.limit,
+      });
+      this.totalNodes = graph.node_count;
+      this.totalEdges = graph.edge_count;
 
-      const edgeQuery = `
-        FOR table in [${graph.edgeTable}]
-          RETURN LENGTH(table)
-      `;
-      const edges = await api.aql(this.workspace, edgeQuery);
+      // eslint-disable-next-line no-underscore-dangle
+      const prelimNodes = nodes.results.map((node) => node._id.substring(0, node._id.indexOf('/')));
+      this.nodeTypes = prelimNodes.filter((node, index) => prelimNodes.indexOf(node) === index);
+      // eslint-disable-next-line no-underscore-dangle
+      this.edgeTypes = edges.results.length > 0 ? [edges.results[0]._id.substring(0, edges.results[0]._id.indexOf('/'))] : [];
```

Comment:
More possibilities for `split()` and `includes()` here.

Reply:
Further, it seems like some of the ID splitting could perhaps be factored out into a function. It might make this type of logic more readable, or maybe not. Thoughts, @AlmightyYakob?

Reply:
See https://github.com/multinet-app/multinet-client/commit/ceedc898a945927e4e277d13583c9eec195e6c64#diff-c702a8a412d06f504ab01fd49432140ee09ea842900696710d609086d3393813L439-R446 and my comment above about replacing indexOf() with includes()